File: src/main/java/moe/xmcn/catsero/I18n.java
package moe.xmcn.catsero;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import moe.xmcn.catsero.utils.Logger;

import java.net.URI;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.ArrayList;

public class I18n {

    private JSONObject i18nJSON;
    private final CatSero INSTANCE = CatSero.INSTANCE;

    public I18n() {
        try {
            i18nJSON = JSON.parseObject(Files.readAllLines(Paths.get(URI.create(
                    INSTANCE.getDataFolder() + "/locale/" + Configuration.getPlugin().getString("locale")
            ))).toString());
        } catch (Exception e) {
            Logger.logCatch(e);
        }
    }

    public String getI18n(ArrayList<String> arr) {

        JSONObject lastObject = new JSONObject();
        String result = "undefined";

        try {
            for (int i = 0; i < arr.size(); i++) {
                if (i != arr.size() - 1) {
                    lastObject = i18nJSON.getJSONObject(arr.get(i));
                } else {
                    result = lastObject.getString(arr.get(i));
                }
            }
        } catch (Exception e) {
            Logger.logWARN("读取I18n文件时发生错误，请检查JSON文件！");
            Logger.logCatch(e);
        }
        return result;
    }

}

File: src/main/kotlin/moe/xmcn/catsero/v3/I18n.kt
package moe.xmcn.catsero.v3

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import moe.xmcn.catsero.v3.util.Logger
import java.nio.file.Files
import java.nio.file.Paths

interface I18n {

    companion object {

        private val lang: String?
            get() = Configuration.pluginConfig.getString("plugin . lang")

        private val messageJSON: JSONObject
            get() = JSON.parseObject(Files.readString(Paths.get("${CatSero.INSTANCE.dataFolder}/lang/$lang/message.json")))
        private val formatJSON: JSONObject
            get() = JSON.parseObject(Files.readString(Paths.get("${CatSero.INSTANCE.dataFolder}/lang/$lang/format.json")))

        /**
         * 获取消息
         * @param s 节点
         * @return 消息
         */
        fun getMessage(s: String): String {
            return read(s, messageJSON)
        }

        /**
         * 获取格式
         * @param s 节点
         * @return 消息
         */
        fun getFormat(s: String): String {
            return read(s, formatJSON)
        }

        private fun read(node: String, data: JSONObject): String {
            return try {
                val nodes = node.split(".")
                var resx = JSONObject()
                for (i in 1 until nodes.size) {
                    resx = data.getJSONObject(nodes[i - 1])
                }
                resx.getString(nodes[nodes.size - 1]) ?: "undefined"
            } catch (e: Exception) {
                Logger.catchEx(e)
                "undefined"
            }
        }

    }

}